### PR TITLE
Implementing the getRawData method

### PR DESCRIPTION
**Pull request by kovacs-tamas**
Hello!

* Type: bug fix | new feature
* Link to issue: https://github.com/phalcon/incubator/issues/880

**In raising this pull request, I confirm the following (please check boxes):**

- [x ] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/incubator/blob/master/CONTRIBUTING.md)?
- [ x] I have checked that another pull request for this purpose does not exist.
- [ ] I wrote some tests for this PR.

Small description of change:
I've made some changes to keep compatibility and be able to work with PHP7.3, Mongodb 4 and Phalcon 3

Thanks

### CODE REVIEW

**Comment by sergeyklay** [comment]
@kovacs-tamas We don't add changes directly to stable branches or `master`. Could you please send this PR to the `3.4.x` branch?

File: Library/Phalcon/Mvc/MongoCollection.php [hunk 1]
```diff
@@ -590,6 +590,77 @@ public function create()
...
+
+        $query = array_intersect_key($data, $keys);
+
+        $success = false;
+
+        $status = $collection->findOneAndUpdate($query,
+            ['$setOnInsert' => $data],
+            ['new' => true, 'upsert' => true]);
+
+        if ($status == null) {
```
Comment (CameronHall) [inline Library/Phalcon/Mvc/MongoCollection.php]: Can you please use strict checks?

File: Library/Phalcon/Session/Adapter/Mongo.php [hunk 2]
```diff
@@ -143,7 +149,7 @@ public function destroy($sessionId = null)
             ]
         );
 
-        return (bool) $remove['ok'];
+        return (bool)$remove['ok'];
```
[the remark this answers is not in the document]
Comment (kovacs-tamas) [reply Library/Phalcon/Session/Adapter/Mongo.php]: Sorry, i guess the PHPStorm's did it. I've fixed.

File: Library/Phalcon/Mvc/MongoCollection.php (hunk 1 above)
Comment (kovacs-tamas) [reply Library/Phalcon/Mvc/MongoCollection.php]: Done

File: Library/Phalcon/Db/Adapter/MongoDB/GridFS/Bucket.php [hunk 3]
```diff
@@ -137,6 +137,26 @@ public function downloadToStream($id, $destination)
...
+     */
+    public function getRawData($id)
+    {
+        $file=$this->collectionWrapper->findFileById($id);
+
+        if ($file===null) {
+            throw FileNotFoundException::byId($id, $this->getFilesNamespace());
+        }
+
+        $stream=new ReadableStream($this->collectionWrapper, $file);
```
[the remark this answers is not in the document]
Comment (kovacs-tamas) [reply Library/Phalcon/Db/Adapter/MongoDB/GridFS/Bucket.php]: Done

File: Library/Phalcon/Db/Adapter/MongoDB/GridFS/Bucket.php [hunk 4]
```diff
@@ -137,6 +137,26 @@ public function downloadToStream($id, $destination)
...
+     * @param mixed    $id File ID
+     *
+     * @return string File content
+     * @throws FileNotFoundException
+     */
+    public function getRawData($id)
+    {
+        $file=$this->collectionWrapper->findFileById($id);
+
+        if ($file===null) {
```
[the remark this answers is not in the document]
Comment (kovacs-tamas) [reply Library/Phalcon/Db/Adapter/MongoDB/GridFS/Bucket.php]: Done

File: Library/Phalcon/Mvc/MongoCollection.php [hunk 5]
```diff
@@ -590,6 +590,77 @@ public function create()
...
+
+        /**
+         * The messages added to the validator are reset here
+         */
+        $this->_errorMessages = [];
+
+        /**
+         * Execute the preSave hook
+         */
+        if ($this->_preSave($this->_dependencyInjector, self::$_disableEvents, $exists) === false) {
```
[the remark this answers is not in the document]
Comment (kovacs-tamas) [reply Library/Phalcon/Mvc/MongoCollection.php]: Done

File: Library/Phalcon/Session/Adapter/Mongo.php [hunk 6]
```diff
@@ -160,7 +166,7 @@ public function gc($maxLifetime)
             )
         );
 
-        $minAgeMongo = new \MongoDate(
+        $minAgeMongo = new \MongoDB\BSON\UTCDateTime(
```
[the remark this answers is not in the document]
Comment (kovacs-tamas) [reply Library/Phalcon/Session/Adapter/Mongo.php]: Done

File: Library/Phalcon/Mvc/MongoCollection.php [hunk 7]
```diff
@@ -590,6 +590,77 @@ public function create()
...
+        if (array_diff_key($keys, $data)) {
+            throw new \Exception("Criteria parameter must be array with one or more attributes of the model");
+        }
+
+        $query = array_intersect_key($data, $keys);
+
+        $success = false;
+
+        $status = $collection->findOneAndUpdate($query,
+            ['$setOnInsert' => $data],
```
[the remark this answers is not in the document]
Comment (kovacs-tamas) [reply Library/Phalcon/Mvc/MongoCollection.php]: Done

**Comment by sergeyklay** [comment]
@kovacs-tamas Could you please submit just the commits that belong to the PR?

**Comment by kovacs-tamas** [comment]
@sergeyklay Theoretically done. Could you please check it?

**Comment by sergeyklay** [comment]
@kovacs-tamas This branch has conflicts that must be resolved. Could you please rebase?

**Comment by kovacs-tamas** [comment]
The conflicts have been solved

File: Library/Phalcon/Db/Adapter/MongoDB/GridFS/Bucket.php [hunk 8]
```diff
@@ -137,6 +137,26 @@ public function downloadToStream($id, $destination)
...
+     * Returns the raw data
+     *
+     * @param mixed    $id File ID
+     *
+     * @return string File content
+     * @throws FileNotFoundException
+     */
+    public function getRawData($id)
+    {
+        $file=$this->collectionWrapper->findFileById($id);
```
Comment (Jeckerson) [inline Library/Phalcon/Db/Adapter/MongoDB/GridFS/Bucket.php]:
```suggestion
        $file = $this->collectionWrapper->findFileById($id);
```

File: Library/Phalcon/Db/Adapter/MongoDB/Database.php [hunk 9]
```diff
@@ -315,4 +315,10 @@ public function withOptions(array $options = [])
 
         return new Database($this->manager, $this->databaseName, $options);
     }
+
+
```
Comment (Jeckerson) [inline Library/Phalcon/Db/Adapter/MongoDB/Database.php]:
```suggestion
```

File: Library/Phalcon/Db/Adapter/MongoDB/GridFS/Bucket.php (hunk 8 above)
Comment (kovacs-tamas) [reply Library/Phalcon/Db/Adapter/MongoDB/GridFS/Bucket.php]: Done

File: Library/Phalcon/Db/Adapter/MongoDB/Database.php (hunk 9 above)
Comment (kovacs-tamas) [reply Library/Phalcon/Db/Adapter/MongoDB/Database.php]: Done

**Comment by sergeyklay** [comment]
Thank you!

**Pull request merged**